modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/CheckpointReadWriteLock.java

```diff
@@ -56,10 +63,17 @@ public class CheckpointReadWriteLock {
      *
      * @param checkpointLock Checkpoint lock.
      * @param throttledLogExecutor Executor for the throttled logger.
+     * @param readLockMetricSource Metrics source.
      */
-    public CheckpointReadWriteLock(ReentrantReadWriteLockWithTracking checkpointLock, Executor throttledLogExecutor) {
+    public CheckpointReadWriteLock(
+            ReentrantReadWriteLockWithTracking checkpointLock,
+            Executor throttledLogExecutor,
+            CheckpointReadWriteLockMetricSource readLockMetricSource
+    ) {
         this.checkpointLock = checkpointLock;
         this.log = Loggers.toThrottledLogger(Loggers.forClass(CheckpointReadWriteLock.class), throttledLogExecutor);
+        this.metrics = readLockMetricSource;
+        this.readLockWaitingThreadsGauge = metrics.readLockWaitingThreadsGauge();
```

Comment:
Not sure why you called it gauge, any hidden meaning? In the context of Ignite metrics framework the gauge metric instance is immutable, because it simply delegates calculations to some other code. For this reason this naming makes me confused